core/src/main/scala/org/http4s/parser/ScalazDeliverySchemes.scala

```diff
@@ -14,10 +14,10 @@ private[http4s] object ScalazDeliverySchemes {
   // scalastyle:off public.methods.have.type
   implicit def Disjunction[L <: HList, Out](implicit unpack: Unpack.Aux[L, Out]) =
     new DeliveryScheme[L] {
-      type Result = ParseFailure \/ Out
+      type Result = ParseError \/ Out
       def success(result: L) = \/-(unpack(result))
-      def parseError(error: ParseError) = -\/(ParseFailure("", errorFormattter.formatExpectedAsString(error)))
-      def failure(error: Throwable) = -\/(ParseFailure("Exception during parsing.", error.getMessage))
+      def parseError(error: ParseError) = -\/(error)
+      def failure(error: Throwable) = throw error
```

Comment:
It depends whether we're looking at `\/` as a `Try` or an `Either`.  Since we're not returning generic `Throwable` on the `Left`, I consider it an `Either`.  And I think any parser that's throwing anything but a `ParseError` is broken.  I think I'm getting more comfortable with this.

I'll leave this open until sometime tomorrow in case someone else wants to step in and call me a fool.